README.md

```diff
@@ -234,99 +223,79 @@ To specify that the variable value is required, add the prefix `!` or `^`.
+|-----------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `'{x=100:number}'`    | Define a variable x of number type and default value 100.                                                                                                                              |
+| `'{x:number}'`        | Define a variable x of number type                                                                                                                                                     |
+| `'{x}'`               | Define a variable x                                                                                                                                                                    |
+| `'{x=100}ABC{y}123'`  | Define two variables x and y. The default value of x is 100. The resolved value will be a concatenation of x, 'ABC', y, and '123'. For example, x=50, y=YYY will produce '50ABCYYY123' |
+| `'{!x}'`              | Define a required variable x                                                                                                                                                           |
+| `'{x=100}ABC{^y}123'` | Define two variables, x and y. The default value of x is 100, and y is required.                                                                                                       |
 
 To use custom methods, configure the REST connector with the `operations` property, which is an array of objects, each of which can have these properties:
 
 - `template` defines the API structure. 
 - `functions` defines JavaScript methods that accept the specified list of parameter names.
 
-```javascript
-var loopback = require("loopback");
-
-var ds = loopback.createDataSource({
-  connector: require("loopback-connector-rest"),
+```ts
+const config = {
+  connector: 'rest',
   debug: false,
   operations: [{
     template: {
-      "method": "GET",
-      "url": "http://maps.googleapis.com/maps/api/geocode/{format=json}",
-      "headers": {
-        "accepts": "application/json",
-        "content-type": "application/json"
+      method: 'GET',
+      url: 'https://maps.googleapis.com/maps/api/geocode/{format=json}',
+      headers: {
+        accepts: 'application/json',
+        'content-type': 'application/json'
       },
-      "query": {
-        "address": "{street},{city},{zipcode}",
-        "sensor": "{sensor=false}"
+      query: {
+        address: '{street},{city},{zipcode}',
+        sensor: '{sensor=false}'
       },
-      "responsePath": "$.results[0].geometry.location"
+      responsePath: '$.results[0].geometry.location'
     },
     functions: {
-      "geocode": ["street", "city", "zipcode"]
+      geocode: ['street', 'city', 'zipcode']
     }
   }]
-});
+}
 ```
 
 Now you can invoke the geocode API in Node.js as follows:
 
+// @TODO: How to do that in LB4?
```

Comment:
Was not able to upgrade from LB3 to LB4 even checking the current code/documentation/issues/....